### PR TITLE
[ISSUE #913] Push docker image to apache repo

### LINKED ISSUE #913: [Enhancement] upload eventmesh docker image into apache repo
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

As is the docker image was uploaded into docker hub, to be will be needed to be uploaded into apache repo.

### Describe the solution you'd like

upload docker image into apache offcial repo.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by kartiktayal**
Fixes #913.

### Motivation

*Currently, the workflow only pushed to eventmesh repo on dockerhub*
*We need the docker image to be pushed under apache repo as well*

### Modifications

*Added the tag for pushing to apache repo on dockerhub*

### Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by kartiktayal** [comment]
Can I please get some reviews on this @Alonexc @mxsm

File: .github/workflows/build.yaml
Comment (Pil0tXia) [inline .github/workflows/build.yaml:42]: I don't think adding `apache` to the script will help solve this issue. Actually this script never works correctly after commit https://github.com/apache/eventmesh/commit/2e6e6bfbdc3ee629eff93ef8804094c88f8bbc6b.

May you please make neccesary modifications and verify it works in your forked repo? Then we can set the DockerHub auth in repo ENVs.

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml
Comment (kartiktayal) [reply .github/workflows/build.yaml:42]: You want me to use my docker creds for my repo and than check if the image is pushed. Is my assumption correct?

Comment (Pil0tXia) [reply .github/workflows/build.yaml:42]: Yes, in this way, we will be able to locate the reason of function failure.

Comment (kartiktayal) [reply .github/workflows/build.yaml:44]: Made the changes to the workflow. The docker image is getting pushed now. Please let me know if we need to update the repos where the image is to be pushed.

**Comment by kartiktayal** [comment]
Request for review @Pil0tXia

**Comment by Pil0tXia** [comment]
I saw your push history, but there was no effective tag was available, which causes an error when pulling.

image

image

**Comment by kartiktayal** [comment]
> I saw your push history, but there was no effective tag available, which causes an error when pulling.
> 
> 
> 
> image
> 
> 
> 
> image
> 
> 

I actually deleted the image after it was pushed. Will trigger the workflow again later today if you need to check the image.

**Comment by Pil0tXia** [comment]
Yes, please let me test it.

**Comment by Pil0tXia** [comment]
You may use the following command to run the docker:

```
sudo docker run -d --name eventmesh-test -p 10000:10000 -p 10105:10105 -p 10205:10205 -p 10106:10106 -v /data/eventmesh/rocketmq/conf/eventmesh.properties:/data/app/eventmes
h/conf/eventmesh.properties -t kartiktayal/eventmesh:latest
```

It started successfully and there was log output:

image

image

I would like to inform you that EventMesh needs the following JVM options to start:

```
-Dlog4j.configurationFile=eventmesh-runtime/conf/log4j2.xml
-Deventmesh.log.home=eventmesh-runtime/logs
-Deventmesh.home=eventmesh-runtime
-DconfPath=eventmesh-runtime/conf
```

You may add these options to the `Dockerfile` which can be configured with `file` field of docker/build-push-action:

image

Some information about Dockerfile: https://stackoverflow.com/questions/57153917/how-to-set-jvm-settings-in-dockerfile.

In addition, we would like to build and push docker images only when there is a new release, such as `1.9.0` and the future releases. This requires you to configure events that trigger workflows, details of which can be found at https://docs.github.com/zh/actions/using-workflows/events-that-trigger-workflows#release , and make some modification on `tags` field to tag them with the current version (if it supports)

Good job. 👍

**Comment by kartiktayal** [comment]
Thank you so much. Will work on building this on release workflow.

**Comment by kartiktayal** [comment]
@Pil0tXia , I hope the changes are as requested.
I used the command 
`docker pull kartiktayal/eventmesh:testing`
and I was able to pull the image using this tag

**Review by Pil0tXia — APPROVED** [review]
image

image

Good job, LGTM.

**Comment by kartiktayal** [comment]
@Alonexc , is this PR good to merge or is there anything pending in this?

**Review by Alonexc — APPROVED** [review]

**Comment by xwm1992** [comment]
The problem of `secrets.DOCKERHUB_USERNAME` and `secrets.DOCKERHUB_PASSWORD` has not been solved yet, temporarily unable to successfully push the image to the eventmesh mirror warehouse, and will continue to follow up this pr.

**Review by xwm1992 — DISMISSED** [review]

**Comment by Pil0tXia** [comment]
Mind before merge: `eventmesh/eventmesh` line should be removed and leave `apache/eventmesh` only to make default secrets work.

**Comment by kartiktayal** [comment]
> Mind before merge: `eventmesh/eventmesh` line should be removed and leave `apache/eventmesh` only to make default secrets work.

Do you want me remove the eventmesh/eventmesh path from the image tag?
Let me know if you need that, will update the PR

**Comment by Pil0tXia** [comment]
@kartiktayal Yes, please. Because the current default repo secret will not be able to support both `apache` and `eventmesh` namespace, and the former is preferred.

**Comment by kartiktayal** [comment]
@Pil0tXia Done.

**Review by xwm1992 — DISMISSED** [review]

**Review by Pil0tXia — APPROVED** [review]

**Comment by Pil0tXia** [comment]
@mxsm This script can be triggered manually after merging to check that our repo's default secret setting is correct.

**Comment by Pil0tXia** [comment]
@xwm1992 @qqeasonchen The dockerhub access of the repo has been acquired. This PR can be merged to test its functionality.

@kartiktayal You may merge the lastest master branch to this PR to pass the CI.

**Comment by kartiktayal** [comment]
> @xwm1992 @qqeasonchen The dockerhub access of the repo has been acquired. This PR can be merged to test its functionality.
> 
> @kartiktayal You may merge the lastest master branch to this PR to pass the CI.

Done.

**Review by xwm1992 — APPROVED** [review]

**Comment by xwm1992** [comment]
@kartiktayal @Pil0tXia 
Here is the eventmesh apache docker hub repo: https://hub.docker.com/r/apache/eventmesh
and I have changed the build.yaml for `username: ${{ secrets.DOCKERHUB_USER }}` & `password: ${{ secrets.DOCKERHUB_TOKEN }}`

**Review by lrhkobe — APPROVED** [review]

**Pull request merged**
